src/main/java/mongodb/Main.java

```diff
@@ -0,0 +1,220 @@
+     *
+     * Here we update the PrepTimeInMinutes value on the document we
+     * just found.
+     */
+    Bson updateFilter = Updates.set("prepTimeInMinutes", 72);
+
+    // The following FindOneAndUpdateOptions specify that we want it to return
+    // the *updated* document to us. By default, we get the document as it was *before*
+    // the update.
+    FindOneAndUpdateOptions options = new FindOneAndUpdateOptions();
+    options.returnDocument(ReturnDocument.AFTER);
+
+    // The updatedDocument object is a Recipe object that reflects the
+    // changes we just made.
+    try {
+      Recipe updatedDocument = collection.findOneAndUpdate(findPotato,
+              updateFilter, options);
+      if (updatedDocument == null) {
+        System.out.println("Couldn't update the recipe.");
+      } else {
+        System.out.println("Updated the recipe to: " + updatedDocument);
+      }
+    } catch (MongoException me) {
+      System.err.println("Could not update any recipes due to an error: " + me);
+    }
+
+    /*      *** DELETE DOCUMENTS ***
+     *
+     *      As with other CRUD methods, you can delete a single document
+     *      or all documents that match a specified filter. To delete all
+     *      of the documents in a collection, pass an empty filter to
+     *      the deleteMany() method. In this example, we'll delete 2 of
+     *      the recipes.
+     */
+    Bson deleteFilter = Filters.in("name", Arrays.asList("elotes", "fried rice"));
+    try {
+      DeleteResult deleteResult = collection
+              .deleteMany(deleteFilter);
+        System.out.println(String.format("Deleted %d documents.", deleteResult.getDeletedCount()));
+    } catch (MongoException me) {
+      System.err.println("Could not delete any recipes due to an error: " + me);
+    }
+
+    // always close the connection when done working with the client
+    mongoClient.close();
+  }
+
+  // helper method with a few pre-wired recipes we can insert into the database as examples.
+  public static List<Recipe> getRecipes() {
+    return Arrays.asList(
+            new Recipe("elotes",
+                    Arrays.asList("corn", "mayonnaise", "cotija cheese", "sour cream", "lime" ),
+                    35),
+            new Recipe("loco moco",
+                    Arrays.asList("ground beef", "butter", "onion", "egg", "bread bun", "mushrooms" ),
+                    54),
+            new Recipe("patatas bravas",
+                    Arrays.asList("potato", "tomato", "olive oil", "onion", "garlic", "paprika" ),
+                    80),
+            new Recipe("fried rice",
+                    Arrays.asList("rice", "soy sauce", "egg", "onion", "pea", "carrot", "sesame oil" ),
+                    40)
+    );
+  }
+
+  // POJO (Plain Old Java Object) class defining a recipe. This class is a POJO because it contains getters and
+  // setters for every member variable as well as an empty constructor.
+  static public class Recipe {
```

Comment:
Question: since this is embedded, do you need `public`?

Nitpick: Seems out of conventional order (visibility is usually before static, e.g. `public static void main()`)

Reply:
Interestingly enough the Java Driver seems to require public here.